### PR TITLE
MVP-13362: Update OpenApi schema for Get Bom response type

### PR DESCRIPTION
**Pull request by smahre**

### CODE REVIEW

**Review by gbhoot — COMMENTED** [review]
Try copying the whole openApi into editor.swagger.io. Noticed a couple syntax errors

File: openapi/v2/Bom/bom-get.yml [hunk 1]
```diff
@@ -0,0 +1,209 @@
+openapi: 3.0.0
+info:
+  version: '3.0'
+  title: Get Bom API
+  license:
+    name: MIT
+servers:
+- url: https://{orgId}.salesforce.com/services/apexrest/{namespace}/v3
```
Comment (gbhoot) [inline openapi/v2/Bom/bom-get.yml]: This is v2 right not v3?

File: openapi/v2/Bom/bom-get.yml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+          type: string
+          format: string
+      - name: level
+        in: query
+        description: Level of BOM to fetch. Defaults to All levels.
+        required: false
+        schema:
+          type: string
+          format: string
+          default: 'all'
```
Comment (gbhoot) [inline openapi/v2/Bom/bom-get.yml]: Looked at the code and were expecting to receive an Integer. Should we specify an number here is required and not string like 'first' or 'second'?
Also in package, we should handle scenario where user passes in 'all' as its default.
@pkjayk what do you think default 'all' or default '0' for all?

**Review by gbhoot — COMMENTED** [review]
First check, left some comments

File: openapi/v2/Bom/bom-get.yml (hunk 1 above)
Comment (pkjayk) [reply openapi/v2/Bom/bom-get.yml]: yes make this v2 if possible.

File: openapi/v2/Bom/bom-get.yml (hunk 2 above)
Comment (pkjayk) [reply openapi/v2/Bom/bom-get.yml]: I think using `all` is better. Can the package expect a string instead?

File: openapi/v2/Bom/bom-get.yml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+          - type
+          - url
+          properties:
+            type:
+              type: string
+            url:
+              type: string
+        Custom_Field__c:
+          type: object
+          description: Placeholder for any custom fields an org may have on the Change__c object.
```
Comment (pkjayk) [inline openapi/v2/Bom/bom-get.yml]: should be `Assembly__c` object

File: openapi/v2/Bom/bom-get.yml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        deltaBom:
+          type: object
+          items:
+            $ref: '#/components/schemas/DeltaBom'
+          description: Contains information about changes in the bom.
+    ReturnBom:
+      type: array
+      items:
+            $ref: '#/components/schemas/ReturnBomItem'
+    ReturnBomItem:
```
Comment (pkjayk) [inline openapi/v2/Bom/bom-get.yml]: @gbhoot @smahre We defined an Assembly schema before, should we reuse that?

File: openapi/v2/Bom/bom-get.yml [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+      properties:
+        Id:
+          type: string
+          description: Id of the associated object
+        attributes:
+          type: object
+          required:
+          - type
+          - url
+          properties:
```
Comment (pkjayk) [inline openapi/v2/Bom/bom-get.yml]: Where'd type and URL come from?

**Review by pkjayk — COMMENTED** [review]
One question I have is what fields are we returning to the user for Assembly?

Is it all fields (I think this should be default behavior)? In the future we can introduce a pattern for specifying fields for endpoints to return.

File: openapi/v2/Bom/bom-get.yml (hunk 2 above)
Comment (smahre) [reply openapi/v2/Bom/bom-get.yml]: @pkjayk So we want it to be able to accept either 'all' or '1', '2', '3', etc?

File: openapi/v2/Bom/bom-get.yml (hunk 4 above)
Comment (smahre) [reply openapi/v2/Bom/bom-get.yml]: @pkjayk The returnBom contains both Assemblies and manufacturer Items in the array. I wanted to create a schema that would work for both of those types. We could separate the two types into two different arrays possibly.

File: openapi/v2/Bom/bom-get.yml (hunk 5 above)
Comment (smahre) [reply openapi/v2/Bom/bom-get.yml]: "attributes": {
                "type": "PLMSM__Manufacturer_Item__c",
                "url": "/services/data/v49.0/sobjects/PLMSM__Manufacturer_Item__c/a0b1U00002oDxGTQA0"
            }

File: openapi/v2/Bom/bom-get.yml (hunk 4 above)
Comment (pkjayk) [reply openapi/v2/Bom/bom-get.yml]: Yes let's do that please and reuse Assembly

**Pull request merged**